providers/go-feature-flag/pkg/provider.go

```diff
@@ -317,7 +317,7 @@ func convertCache[T model.JsonType](value interface{}) (model.GenericResolutionD
 
 // evaluateWithRelayProxy is calling GO Feature Flag relay proxy to evaluate the file.
 func evaluateWithRelayProxy[T model.JsonType](provider *Provider, ctx context.Context, goffRequestBody model.EvalFlagRequest, flagName string, defaultValue T) model.GenericResolutionDetail[T] {
-	cacheKey := fmt.Sprintf("%s-%s", flagName, goffRequestBody.User.Key)
+	cacheKey := fmt.Sprintf("%s-%+v", flagName, goffRequestBody.User)
```

Comment:
```suggestion
	cacheKey := fmt.Sprintf("%s-%+v", flagName, goffRequestBody.EvaluationContext)
```

This is my mistake here, but we should better use `EvaluationContext` than user, because `User` should be deprecated.